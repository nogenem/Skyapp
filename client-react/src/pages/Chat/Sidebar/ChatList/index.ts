export { default as ChatList, UnconnectedChatList } from './ChatList';
export type { TProps } from './ChatList';
