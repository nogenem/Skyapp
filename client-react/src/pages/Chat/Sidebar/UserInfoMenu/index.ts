export {
  default as UserInfoMenu,
  UnconnectedUserInfoMenu,
} from './UserInfoMenu';
export type { TProps } from './UserInfoMenu';
