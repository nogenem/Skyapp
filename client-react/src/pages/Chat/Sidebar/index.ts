export { default as Sidebar } from './Sidebar';
export type { TProps } from './Sidebar';
