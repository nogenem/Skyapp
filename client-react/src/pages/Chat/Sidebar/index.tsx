export { default as Sidebar, UnconnectedSidebar } from './Sidebar';
export type { TProps } from './Sidebar';
