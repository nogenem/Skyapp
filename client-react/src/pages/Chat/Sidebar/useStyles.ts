import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    container: {
      minWidth: '330px',
      maxWidth: '330px',
      height: '100vh',
      padding: theme.spacing(1),

      '&.expanded': {
        width: '100%',
        maxWidth: 'unset',
      },
      '&.hidden': {
        display: 'none',
      },
    },
    btnsContainer: {
      display: 'flex',
      alignItems: 'center',
      justifyContent: 'flex-end',
      marginTop: theme.spacing(1),
    },
  }),
);

export default useStyles;
