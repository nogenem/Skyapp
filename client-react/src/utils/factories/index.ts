import * as models from './models';
import * as states from './states';

export default {
  models,
  states,
};
