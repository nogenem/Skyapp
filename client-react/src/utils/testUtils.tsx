import React from 'react';
import { Provider } from 'react-redux';

import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { IAppState } from '~/redux/store';

export const getMockStore = () => {
  const middlewares = [thunk];
  return configureStore(middlewares);
};

export const getRenderWithRedux = () => {
  const emptyState: Partial<IAppState> = {};
  const mockStore = getMockStore();

  return (
    ui: React.ReactNode,
    initialState: Partial<IAppState> = emptyState,
  ) => {
    const store = mockStore(initialState);
    return {
      ...render(<Provider store={store}>{ui}</Provider>),
    };
  };
};
