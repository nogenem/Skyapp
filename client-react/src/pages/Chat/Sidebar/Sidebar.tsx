import React from 'react';
import { useTranslation } from 'react-i18next';

import { Paper, IconButton } from '@material-ui/core';
import {
  PersonAdd as PersonAddIcon,
  GroupAdd as GroupAddIcon,
} from '@material-ui/icons';

import { TextInput, NewChatModal, NewGroupModal } from '~/components';

import { ChatList } from '../ChatList';
import { UserInfoMenu } from '../UserInfoMenu';
import useStyles from './useStyles';

interface IOwnProps {
  isUserEmailConfirmed: boolean;
}

type TProps = IOwnProps;

const Sidebar = ({ isUserEmailConfirmed }: TProps) => {
  const [isNewChatModalOpen, setIsNewChatModalOpen] = React.useState(false);
  const [isNewGroupModalOpen, setIsNewGroupModalOpen] = React.useState(false);
  const [filter, setFilter] = React.useState('');
  const { t: trans } = useTranslation(['Common']);
  const classes = useStyles();

  const handleNewChatModalOpen = () => {
    setIsNewChatModalOpen(true);
  };

  const handleNewChatModalClose = () => {
    setIsNewChatModalOpen(false);
  };

  const handleNewGroupModalOpen = () => {
    setIsNewGroupModalOpen(true);
  };

  const handleNewGroupModalClose = () => {
    setIsNewGroupModalOpen(false);
  };

  const handleFilterChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    setFilter(e.target.value.toLowerCase());
  };

  return (
    <>
      <Paper square className={classes.container} elevation={8}>
        <UserInfoMenu />

        {isUserEmailConfirmed && (
          <>
            <TextInput
              id="chat-sidebar-search"
              name="search"
              label={trans('Common:Search')}
              autoComplete="search"
              type="text"
              fullWidth
              variant="outlined"
              margin="dense"
              onChange={handleFilterChange}
            />

            <div className={classes.btnsContainer}>
              <IconButton
                aria-label={trans('Common:New Chat')}
                title={trans('Common:New Chat')}
                onClick={handleNewChatModalOpen}
              >
                <PersonAddIcon />
              </IconButton>
              <IconButton
                aria-label={trans('Common:New Group')}
                title={trans('Common:New Group')}
                onClick={handleNewGroupModalOpen}
              >
                <GroupAddIcon />
              </IconButton>
            </div>

            <ChatList filter={filter} />
          </>
        )}
      </Paper>
      <NewChatModal
        isOpen={isNewChatModalOpen}
        onClose={handleNewChatModalClose}
      />
      <NewGroupModal
        isOpen={isNewGroupModalOpen}
        onClose={handleNewGroupModalClose}
      />
    </>
  );
};

export type { TProps };
export default Sidebar;
