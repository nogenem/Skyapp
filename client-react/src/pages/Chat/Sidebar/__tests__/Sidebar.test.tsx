import React from 'react';

import { getRenderWithRedux } from '~/utils/testUtils';

import Sidebar from '../Sidebar';

const renderWithRedux = getRenderWithRedux();

describe('Sidebar', () => {
  it('renders correctly when `isUserEmailConfirmed` is true', () => {
    const { container } = renderWithRedux(
      <Sidebar
        isUserEmailConfirmed={true}
        isSmall={false}
        activeChannelId={undefined}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly when `isUserEmailConfirmed` is false', () => {
    const { container } = renderWithRedux(
      <Sidebar
        isUserEmailConfirmed={false}
        isSmall={false}
        activeChannelId={undefined}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly when `isSmall` is true and `activeChannelId` is falsy', () => {
    const { container } = renderWithRedux(
      <Sidebar
        isUserEmailConfirmed={false}
        isSmall={true}
        activeChannelId={undefined}
      />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly when `isSmall` is true and `activeChannelId` is truthy', () => {
    const { container } = renderWithRedux(
      <Sidebar
        isUserEmailConfirmed={false}
        isSmall={true}
        activeChannelId={'1'}
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
