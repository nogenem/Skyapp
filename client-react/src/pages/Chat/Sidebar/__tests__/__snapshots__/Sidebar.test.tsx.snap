// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar renders correctly when \`isSmall\` is true and \`activeChannelId\` is falsy 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-container-59 expanded MuiPaper-elevation8"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root makeStyles-container-61 MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="makeStyles-container-63"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-64"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </div>
        <span
          class="makeStyles-dotContainer-65"
        >
          <span
            class="makeStyles-dot-66 makeStyles-dot_active-67 "
            title="Common:Active"
          />
        </span>
      </div>
      <div
        class="makeStyles-textContainer-62"
      >
        <span
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit MuiTypography-noWrap"
          title=""
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`Sidebar renders correctly when \`isSmall\` is true and \`activeChannelId\` is truthy 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-container-85 hidden MuiPaper-elevation8"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root makeStyles-container-87 MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="makeStyles-container-89"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-90"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </div>
        <span
          class="makeStyles-dotContainer-91"
        >
          <span
            class="makeStyles-dot-92 makeStyles-dot_active-93 "
            title="Common:Active"
          />
        </span>
      </div>
      <div
        class="makeStyles-textContainer-88"
      >
        <span
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit MuiTypography-noWrap"
          title=""
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`Sidebar renders correctly when \`isUserEmailConfirmed\` is false 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-container-33  MuiPaper-elevation8"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root makeStyles-container-35 MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="makeStyles-container-37"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-38"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </div>
        <span
          class="makeStyles-dotContainer-39"
        >
          <span
            class="makeStyles-dot-40 makeStyles-dot_active-41 "
            title="Common:Active"
          />
        </span>
      </div>
      <div
        class="makeStyles-textContainer-36"
      >
        <span
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit MuiTypography-noWrap"
          title=""
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>
</div>
`;

exports[`Sidebar renders correctly when \`isUserEmailConfirmed\` is true 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-container-1  MuiPaper-elevation8"
  >
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root makeStyles-container-3 MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="makeStyles-container-5"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-6"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
            />
          </svg>
        </div>
        <span
          class="makeStyles-dotContainer-7"
        >
          <span
            class="makeStyles-dot-8 makeStyles-dot_active-9 "
            title="Common:Active"
          />
        </span>
      </div>
      <div
        class="makeStyles-textContainer-4"
      >
        <span
          class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit MuiTypography-noWrap"
          title=""
        />
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      class="MuiFormControl-root MuiTextField-root makeStyles-textInput-20 MuiFormControl-marginDense MuiFormControl-fullWidth"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
        for="chat-sidebar-search"
        id="chat-sidebar-search-label"
      >
        Common:Search
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          autocomplete="search"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="chat-sidebar-search"
          name="search"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-21 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-23"
          >
            <span>
              Common:Search
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="makeStyles-btnsContainer-2"
    >
      <button
        aria-label="Common:New Chat"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        title="Common:New Chat"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm-9-2V7H4v3H1v2h3v3h2v-3h3v-2H6zm9 4c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <button
        aria-label="Common:New Group"
        class="MuiButtonBase-root MuiIconButton-root"
        tabindex="0"
        title="Common:New Group"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M8 10H5V7H3v3H0v2h3v3h2v-3h3v-2zm10 1c1.66 0 2.99-1.34 2.99-3S19.66 5 18 5c-.32 0-.63.05-.91.14.57.81.9 1.79.9 2.86s-.34 2.04-.9 2.86c.28.09.59.14.91.14zm-5 0c1.66 0 2.99-1.34 2.99-3S14.66 5 13 5c-1.66 0-3 1.34-3 3s1.34 3 3 3zm6.62 2.16c.83.73 1.38 1.66 1.38 2.84v2h3v-2c0-1.54-2.37-2.49-4.38-2.84zM13 13c-2 0-6 1-6 3v2h12v-2c0-2-4-3-6-3z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <h6
      class="MuiTypography-root makeStyles-chatsText-25 MuiTypography-subtitle2"
    >
      Common:CHATS
    </h6>
    <ul
      class="MuiList-root MuiList-padding"
    />
  </div>
</div>
`;
